### ISSUE TITLE
Uncaught TypeError: undefined is not a function

### ISSUE THREAD

**Issue by ELD** [lloeki/ex-mode#39]
[Enter steps to reproduce below:]

1. Hit the `esc` key
2. Type `w` to write the file

**Atom Version**: 0.194.0 :warning: **in 1.0 API Preview Mode** :warning:
**System**: Mac OS X 10.10.3
**Thrown From**: [ex-mode](https://github.com/lloeki/ex-mode) package, v0.4.1

### Stack Trace

Uncaught TypeError: undefined is not a function

```
At /Users/eric/.atom/packages/ex-mode/lib/ex-state.coffee:56

TypeError: undefined is not a function
  at Ex.write (/Users/eric/.atom/packages/ex-mode/lib/ex.coffee:117:29)
  at Ex.w (/Users/eric/.atom/packages/ex-mode/lib/ex.coffee:143:5)
  at /Users/eric/.atom/packages/ex-mode/lib/ex.coffee:1:1
  at Command.execute (/Users/eric/.atom/packages/ex-mode/lib/command.coffee:148:7)
  at ExState.processOpStack (/Users/eric/.atom/packages/ex-mode/lib/ex-state.coffee:49:17)
  at ExState.pushOperations (/Users/eric/.atom/packages/ex-mode/lib/ex-state.coffee:40:6)
  at ExViewModel.ViewModel.confirm (/Users/eric/.atom/packages/ex-mode/lib/view-model.coffee:12:14)
  at ExViewModel.module.exports.ExViewModel.confirm (/Users/eric/.atom/packages/ex-mode/lib/ex-view-model.coffee:35:5)
  at ExViewModel.confirm (/Users/eric/.atom/packages/ex-mode/lib/ex-view-model.coffee:1:1)
  at ExCommandModeInputView.module.exports.ExCommandModeInputView.confirm (/Users/eric/.atom/packages/ex-mode/lib/ex-command-mode-input-view.coffee:46:16)
  at atom-text-editor.<anonymous> (/Users/eric/.atom/packages/ex-mode/lib/ex-command-mode-input-view.coffee:1:1)
  at atom-text-editor.jQuery.event.dispatch (/Applications/Atom.app/Contents/Resources/app.asar/node_modules/jquery/dist/jquery.js:4430:9)
  at atom-text-editor.elemData.handle (/Applications/Atom.app/Contents/Resources/app.asar/node_modules/jquery/dist/jquery.js:4116:28)
  at KeymapManager.module.exports.KeymapManager.dispatchCommandEvent (/Applications/Atom.app/Contents/Resources/app.asar/node_modules/atom-keymap/lib/keymap-manager.js:519:16)
  at KeymapManager.module.exports.KeymapManager.handleKeyboardEvent (/Applications/Atom.app/Contents/Resources/app.asar/node_modules/atom-keymap/lib/keymap-manager.js:354:22)
  at HTMLDocument.module.exports.WindowEventHandler.onKeydown (/Applications/Atom.app/Contents/Resources/app.asar/src/window-event-handler.js:178:20)

```

### Commands

```
     -1:01.1.0 vim-mode:activate-command-mode (atom-text-editor.editor.cursor-block.vim-mode.command-mode.is-focused)
     -1:00.7.0 ex-mode:open (atom-text-editor.editor.cursor-block.vim-mode.command-mode.is-focused)
     -1:00.5.0 core:confirm (atom-text-editor.editor.mini)
     -0:58.1.0 ex-mode:open (atom-text-editor.editor.cursor-block.vim-mode.command-mode.is-focused)
     -0:57.9.0 core:confirm (atom-text-editor.editor.mini)
     -0:56.1.0 editor:consolidate-selections (atom-text-editor.editor.mini)
     -0:56.1.0 core:cancel (atom-text-editor.editor.mini)
     -0:54.9.0 ex-mode:open (atom-text-editor.editor.cursor-block.vim-mode.command-mode.is-focused)
     -0:54.7.0 core:confirm (atom-text-editor.editor.mini)
     -0:53.5.0 core:save (atom-text-editor.editor.cursor-block.vim-mode.command-mode.is-focused)
     -0:53.1.0 core:close (atom-text-editor.editor.cursor-block.vim-mode.command-mode.is-focused)
     -0:51.1.0 settings-view:open (atom-text-editor.editor.cursor-block.vim-mode.command-mode)
  3x -0:46.7.0 core:confirm (atom-text-editor.editor.mini)
     -0:03.2.0 core:close (atom-text-editor.editor.mini)
     -0:01.8.0 ex-mode:open (atom-text-editor.editor.cursor-block.vim-mode.command-mode)
     -0:01.0 core:confirm (atom-text-editor.editor.mini.is-focused)
```

### Config

```json
{
  "core": {
    "themes": [
      "one-dark-ui",
      "solarized-dark-syntax"
    ],
    "followSymlinks": true
  }
}
```

### Installed Packages

```coffee
# User
advanced-new-file, v0.4.0
atom-grails, v0.1.0
autocomplete-paths, v1.0.2
autocomplete-plus, v2.12.0
block-cursor, v0.12.4
cargo-test-runner, v0.0.3
ex-mode, v0.4.1
language-groovy, v0.4.0
language-latex, v0.6.1
language-rust, v0.4.3
latex, v0.22.0
linter-clang, v2.25.1
linter-rust, v0.0.9
minimap, v4.7.6
minimap-find-and-replace, v4.2.0
minimap-selection, v4.2.0
project-manager, v1.15.5
theme-switcher, v1.1.0
vim-mode, v0.44.0
vim-surround, v0.5.1

# Dev
No dev packages
```

**Comment by lloeki** [comment]
fixed by c43b2658

**Comment by jbiason** [comment]
The problem just happened to me. It could be related to the fact that 1) it was a new "buffer" and/or 2) I used "w file.xml".

**Comment by WhiteVoidia** [comment]
I also got this error when running `e asdf` or `edit asdf` the first time from any file (if it's ran again in the same file it doesn't show an error but the command doesn't do anything)

```
TypeError: undefined is not a function
  at getFullPath (/home/jluke/.atom/packages/ex-mode/lib/ex.coffee:35:33)
  at Ex.edit (/home/jluke/.atom/packages/ex-mode/lib/ex.coffee:102:20)
  at Ex.e (/home/jluke/.atom/packages/ex-mode/lib/ex.coffee:105:19)
  at /home/jluke/.atom/packages/ex-mode/lib/ex.coffee:1:1
  at Command.execute (/home/jluke/.atom/packages/ex-mode/lib/command.coffee:153:7)
  at ExState.processOpStack (/home/jluke/.atom/packages/ex-mode/lib/ex-state.coffee:50:17)
  at ExState.pushOperations (/home/jluke/.atom/packages/ex-mode/lib/ex-state.coffee:40:6)
  at ExViewModel.ViewModel.confirm (/home/jluke/.atom/packages/ex-mode/lib/view-model.coffee:13:14)
  at ExViewModel.module.exports.ExViewModel.confirm (/home/jluke/.atom/packages/ex-mode/lib/ex-view-model.coffee:35:5)
  at ExViewModel.confirm (/home/jluke/.atom/packages/ex-mode/lib/ex-view-model.coffee:1:1)
  at ex-command-mode-input.ExCommandModeInputElement.confirm (/home/jluke/.atom/packages/ex-mode/lib/ex-command-mode-input-element.coffee:46:16)
  at CommandRegistry.module.exports.CommandRegistry.handleCommandEvent (/usr/share/atom/resources/app.asar/src/command-registry.js:241:29)
  at /usr/share/atom/resources/app.asar/src/command-registry.js:3:61
  at KeymapManager.module.exports.KeymapManager.dispatchCommandEvent (/usr/share/atom/resources/app.asar/node_modules/atom-keymap/lib/keymap-manager.js:520:16)
  at KeymapManager.module.exports.KeymapManager.handleKeyboardEvent (/usr/share/atom/resources/app.asar/node_modules/atom-keymap/lib/keymap-manager.js:355:22)
  at HTMLDocument.module.exports.WindowEventHandler.onKeydown (/usr/share/atom/resources/app.asar/src/window-event-handler.js:180:20)
```

**Comment by mleeds95** [comment]
This happened to me as well when I typed ":w file" using Atom 1.0.2 and ex-mode 0.5.1 and vim-mode 0.57.0 on Ubuntu 15.04. Stacktrace:
```
TypeError: undefined is not a function
  at getFullPath (/home/matthew/.atom/packages/ex-mode/lib/ex.coffee:35:33)
  at Ex.write (/home/matthew/.atom/packages/ex-mode/lib/ex.coffee:131:20)
  at Ex.w (/home/matthew/.atom/packages/ex-mode/lib/ex.coffee:143:5)
  at /home/matthew/.atom/packages/ex-mode/lib/ex.coffee:1:1
  at Command.execute (/home/matthew/.atom/packages/ex-mode/lib/command.coffee:153:7)
  at ExState.processOpStack (/home/matthew/.atom/packages/ex-mode/lib/ex-state.coffee:50:17)
  at ExState.pushOperations (/home/matthew/.atom/packages/ex-mode/lib/ex-state.coffee:40:6)
  at ExViewModel.ViewModel.confirm (/home/matthew/.atom/packages/ex-mode/lib/view-model.coffee:13:14)
  at ExViewModel.module.exports.ExViewModel.confirm (/home/matthew/.atom/packages/ex-mode/lib/ex-view-model.coffee:35:5)
  at ExViewModel.confirm (/home/matthew/.atom/packages/ex-mode/lib/ex-view-model.coffee:1:1)
  at ex-command-mode-input.ExCommandModeInputElement.confirm (/home/matthew/.atom/packages/ex-mode/lib/ex-command-mode-input-element.coffee:46:16)
  at CommandRegistry.module.exports.CommandRegistry.handleCommandEvent (/usr/local/share/atom/resources/app.asar/src/command-registry.js:241:29)
  at /usr/local/share/atom/resources/app.asar/src/command-registry.js:3:61
  at KeymapManager.module.exports.KeymapManager.dispatchCommandEvent (/usr/local/share/atom/resources/app.asar/node_modules/atom-keymap/lib/keymap-manager.js:520:16)
  at KeymapManager.module.exports.KeymapManager.handleKeyboardEvent (/usr/local/share/atom/resources/app.asar/node_modules/atom-keymap/lib/keymap-manager.js:355:22)
  at HTMLDocument.module.exports.WindowEventHandler.onKeydown (/usr/local/share/atom/resources/app.asar/src/window-event-handler.js:177:20)
```

**Comment by mleeds95** [comment]
fixed this with 656ed90f7e5391ce7228f9da00f77423573403c2
